vms/components/fee/state_test.go

```diff
@@ -0,0 +1,162 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package fee
+
+import (
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_State_AdvanceTime(t *testing.T) {
+	tests := []struct {
+		name               string
+		initial            State
+		maxGasCapacity     Gas
+		maxGasPerSecond    Gas
+		targetGasPerSecond Gas
+		duration           uint64
+		expected           State
+	}{
+		{
+			name: "cap capacity",
+			initial: State{
+				Capacity: 10,
+				Excess:   0,
```

Comment:
Hmmm the function that is being tested _does_ modify both of these fields though. I feel like it would be weird to split the tests for this function into the 2 components.